src/Polly.RateLimiting/RateLimiterRejectedException.cs

```diff
@@ -0,0 +1,112 @@
+#if !NETCOREAPP
+using System.Runtime.Serialization;
+#endif
+using System.Threading.RateLimiting;
+
+namespace Polly.RateLimiting;
+
+/// <summary>
+/// Exception thrown when a rate limiter rejects an execution.
+/// </summary>
+#if !NETCOREAPP
+[Serializable]
+#endif
+public class RateLimiterRejectedException : ExecutionRejectedException
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RateLimiterRejectedException"/> class.
+    /// </summary>
+    public RateLimiterRejectedException()
+        : base("The operation couldn't be executed because it was rejected by the rate limiter.")
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RateLimiterRejectedException"/> class.
+    /// </summary>
+    /// <param name="retryAfter">The retry after value.</param>
+    public RateLimiterRejectedException(TimeSpan retryAfter)
+        : base($"The operation couldn't be executed because it was rejected by the rate limiter. It can be retried after '{retryAfter}'.")
```

Comment:
```suggestion
        : base($"The operation could not be executed because it was rejected by the rate limiter. It can be retried after '{retryAfter}'.")
```